### PR TITLE
[`test`] attempt to fix CI test for PT 2.0

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR fixes the failing tests related to Pytorch 2.0 upgrade

In Pytorch 2.0, the learning rate schedulers have been re-designed, thus we need a bit of refactoring on `trl` to deal with unexpected behaviors.

In Pytorch 2.0 it seems that the class `_LRScheduler` have been deprecated in favor of `LRScheduler`

## In Pytorch 2.0:

```python
# Including _LRScheduler for backwards compatibility
# Subclass instead of assign because we want __name__ of _LRScheduler to be _LRScheduler (assigning would make it LRScheduler).
class _LRScheduler(LRScheduler):
    pass
```

## In Pytorch < 2.0:

```python

class LambdaLR(_LRScheduler):
    """Sets the learning rate of each parameter group to the initial lr
    times a given function. When last_epoch=-1, sets initial lr as lr.

    Args:"""
```

Hence the check `isinstance(lr_scheduler, torch.optim.lr_scheduler._LRScheduler)` does not pass anymore with Pytorch >=2.0

### CODE REVIEW

File: trl/import_utils.py [hunk 1]
```diff
@@ -12,7 +12,26 @@
...
+def is_torch_greater_2_0():
+    if _is_python_greater_3_8:
+        from importlib.metadata import version
+
+        torch_version = version("torch")
+    else:
+        import pkg_resources
+
+        torch_version = pkg_resources.get_distribution("torch").version
+    return torch_version >= "2.0"
```
Comment (younesbelkada) [inline trl/import_utils.py:37]: it is a way I have found to do versions checking wrt to python version, I would be curious to see if you know better techniques.

File: tests/test_ppo_trainer.py [hunk 2]
```diff
@@ -265,7 +265,10 @@ def test_ppo_step_with_no_ref_sgd_lr_scheduler(self):
         dummy_dataloader = ppo_trainer.dataloader
 
         self.assertTrue(isinstance(ppo_trainer.optimizer.optimizer, torch.optim.SGD))
-        self.assertTrue(isinstance(ppo_trainer.lr_scheduler.scheduler, torch.optim.lr_scheduler.ExponentialLR))
+        if is_torch_greater_2_0():
+            self.assertTrue(isinstance(ppo_trainer.lr_scheduler, torch.optim.lr_scheduler.ExponentialLR))
+        else:
+            self.assertTrue(isinstance(ppo_trainer.lr_scheduler.scheduler, torch.optim.lr_scheduler.ExponentialLR))
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply tests/test_ppo_trainer.py:270]: Tracking the issue at: https://github.com/huggingface/accelerate/issues/1200

**Review by lvwerra — APPROVED** [review]
